python3/raygun4py/raygunprovider.py

```diff
@@ -295,42 +295,42 @@ def _create_message(
         )
 
     def _transform_message(self, message):
+        message = message.copy()
         message = utilities.ignore_exceptions(self.ignored_exceptions, message)
 
         if message is not None:
             message = utilities.filter_keys(self.filtered_keys, message)
-            message["details"]["groupingKey"] = utilities.execute_grouping_key(
+            message.get_details()["groupingKey"] = utilities.execute_grouping_key(
```

Comment:
`message` should be a `RaygunMessage` object and not a `dict`, access properties using accessors.